libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -215,7 +310,48 @@ AP_BattMonitor::init()
             // there will be a gap, but as we always check for drivers[instances] being nullptr
             // this is safe
             _num_instances = instance + 1;
+
+            // Convert the old analog & Bus parameters to the new dynamic parameter groups
+            convert_dynamic_param_groups(instance);
+        }
+    }
+}
+
+void AP_BattMonitor::convert_dynamic_param_groups(uint8_t instance)
+{
+    AP_Param::ConversionInfo info;
+    if (!AP_Param::find_top_level_key_by_pointer(this, info.old_key)) {
```

Comment:
start with looking up the BATTn_TYPE param, and offset from that would be safer

Reply:
Done. I did try to find the index using pointers instead , but no appropriate functions exists to do that I could find. I did try modifying the existing find_key_by_group to do that but the structure of the param table eludes me to generalize the pointer math.